### PR TITLE
fix token storage for access

### PR DESCRIPTION
**Pull request by oallain**
| Q             | A
| ------------- | ---
| Bug fix?      | yes
| New feature?  | no
| BC breaks?   | no
| Fixed issue |

### CODE REVIEW

File: src/Checker/AdminChecker.php [hunk 1]
```diff
@@ -32,10 +32,32 @@ public function isMaintenance(MaintenanceConfiguration $configuration, Request $
         /** @var string $adminPrefix */
         $adminPrefix = $this->params->get('sylius_admin.path_name');
 
+        if (str_starts_with($getRequestUri, '/_profiler') || str_starts_with($getRequestUri, '/_wdt')) {
```
Comment (Jibbarth) [inline src/Checker/AdminChecker.php]: Shouldn't we create a ProfilerChecker instead ?

File: src/Checker/AdminChecker.php [hunk 2]
```diff
@@ -32,10 +32,32 @@ public function isMaintenance(MaintenanceConfiguration $configuration, Request $
...
+            return IsMaintenanceVoterInterface::ACCESS_GRANTED;
+        }
+
         if (false !== mb_strpos($getRequestUri, $adminPrefix, 1)) {
-            if ($this->requestStack->getMainRequest() === $this->requestStack->getCurrentRequest()) {
+            $session = null;
+            if (method_exists($this->requestStack, 'isMainRequest')) {
```
Comment (Jibbarth) [inline src/Checker/AdminChecker.php]: This method has never existed ? I think it 's on RequestEvent instead

File: src/Checker/AdminChecker.php [hunk 3]
```diff
@@ -32,10 +32,32 @@ public function isMaintenance(MaintenanceConfiguration $configuration, Request $
...
+                if (!$this->requestStack->isMainRequest()) {
+                    return IsMaintenanceVoterInterface::ACCESS_DENIED;
+                }
+
+                $session = $this->requestStack->getMainRequest()?->getSession();
+            }
+
+            /** @TODO Drop after remove Symfony 4.4 compatibility */
+            if (method_exists($this->requestStack, 'isMasterRequest')) {
+                if (!$this->requestStack->isMasterRequest()) {
```
Comment (Jibbarth) [inline src/Checker/AdminChecker.php]: same, doesn't exist ?

File: src/Checker/AdminChecker.php (hunk 1 above)
Comment (oallain) [reply src/Checker/AdminChecker.php]: Good idea ;)

File: src/Checker/AdminChecker.php (hunk 2 above)
Comment (oallain) [reply src/Checker/AdminChecker.php]: I fixed it, thanks

**Review by TheGrimmChester — DISMISSED** [review]

File: src/Checker/AdminChecker.php [hunk 4]
```diff
@@ -31,12 +31,14 @@ public function isMaintenance(MaintenanceConfiguration $configuration, Request $
         $getRequestUri = $request->getRequestUri();
         /** @var string $adminPrefix */
         $adminPrefix = $this->params->get('sylius_admin.path_name');
+        $adminPrefix = \DIRECTORY_SEPARATOR . $adminPrefix;
 
-        if (false !== mb_strpos($getRequestUri, $adminPrefix, 1)) {
+        if (str_starts_with($getRequestUri, $adminPrefix)) {
             if ($this->requestStack->getMainRequest() === $this->requestStack->getCurrentRequest()) {
```
Comment (Jibbarth) [inline src/Checker/AdminChecker.php]: what about this `getMainRequest` ?

File: src/Checker/AdminChecker.php [hunk 5]
```diff
@@ -31,12 +31,14 @@ public function isMaintenance(MaintenanceConfiguration $configuration, Request $
...
-                /** @var FlashBagInterface $flashBag */
                 $flashBag = $this->requestStack->getSession()->getBag('flashes');
```
Comment (Jibbarth) [inline src/Checker/AdminChecker.php]: :warning: no getSession on sf 4.4

File: src/EventSubscriber/MaintenanceEventsubscriber.php [hunk 6]
```diff
@@ -47,13 +45,9 @@ public function handle(RequestEvent $event): void
...
         }
 
-        $responseContent = $this->translator->trans('maintenance.ui.message');
-
-        if ('' !== $configuration->getCustomMessage()) {
-            $responseContent = $this->twig->render('@SynoliaSyliusMaintenancePlugin/maintenance.html.twig', [
-                'custom_message' => $configuration->getCustomMessage(),
-            ]);
-        }
+        $responseContent = $this->twig->render('@SynoliaSyliusMaintenancePlugin/maintenance.html.twig', [
```
Comment (Jibbarth) [inline src/EventSubscriber/MaintenanceEventsubscriber.php:48]: this wasn't done in another PR ? Or maybe you merged it in this branch ?

File: src/Checker/AdminChecker.php (hunk 4 above)
Comment (oallain) [reply src/Checker/AdminChecker.php]: My bad ... i've fixed it ;)

File: src/Checker/AdminChecker.php (hunk 5 above)
Comment (oallain) [reply src/Checker/AdminChecker.php]: I've fixed it too.

File: src/EventSubscriber/MaintenanceEventsubscriber.php (hunk 6 above)
Comment (oallain) [reply src/EventSubscriber/MaintenanceEventsubscriber.php:48]: Yes, the PR #32 was merged on this PR.

**Review by TheGrimmChester — APPROVED** [review]

**Review by Jibbarth — APPROVED** [review]
